ScoutSuite/providers/azure/authentication_strategy.py

```diff
@@ -191,31 +124,22 @@ def authenticate(self,
                 client_id = data.get('clientId')
                 client_secret = data.get('clientSecret')
 
-                arm_credentials = ServicePrincipalCredentials(
+                identity_credentials = ClientSecretCredential(
                     client_id=client_id,
-                    secret=client_secret,
-                    tenant=tenant_id
-                )
-
-                aad_graph_credentials = ServicePrincipalCredentials(
-                    client_id=client_id,
-                    secret=client_secret,
-                    tenant=tenant_id,
-                    resource='https://graph.windows.net'
+                    client_secret=client_secret,
+                    tenant_id=tenant_id
                 )
 
             elif msi:
-
-                arm_credentials = MSIAuthentication()
-                aad_graph_credentials = MSIAuthentication(resource='https://graph.windows.net')
+                identity_credentials = ManagedIdentityCredential()
 
             else:
                 raise AuthenticationException('Unknown authentication method')
 
-            return AzureCredentials(arm_credentials,
-                                    aad_graph_credentials,
-                                    tenant_id, subscription_id,
-                                    context)
+            return AzureCredentials(
+                identity_credentials,
+                tenant_id, subscription_id,
+                context)
 
         except Exception as e:
             if ', AdalError: Unsupported wstrust endpoint version. ' \
```

Comment:
Can you check if this still handles https://github.com/nccgroup/ScoutSuite/issues/742?

Reply:
Not sure how to reproduce this error because we do not get an exception with our accounts. But normally, this is an adal Error and we are not using adal anymore, so this exception should not happen and we should remove this.

Reply:
Do you have a Microsoft account (e.g. @hotmail address) you can use? If not I'll test it out.

Reply:
An exception is thrown when logging in with microsoft accounts, it is now handled properly